<?xml version="1.0"?>
<launch>
  <arg name="pose_sources" default="" description=""/>

  <node pkg="pose_estimator_manager" exec="pose_estimator_manager" name="pose_estimator_manager" output="screen">
    <param name="pcd_occupancy_rule/pcd_density_upper_threshold" value="25"/>
    <param name="pcd_occupancy_rule/pcd_density_lower_threshold" value="20"/>
    <param name="pose_sources" value="$(var pose_sources)"/>
    <param name="ar_marker_rule/ar_marker_available_distance" value="30"/>

    <!-- artag -->
    <remap from="~/input/artag/image" to="/sensing/camera/traffic_light/image_raw"/>
    <remap from="~/output/artag/image" to="/sensing/camera/traffic_light/image_raw/relay"/>
    <!-- yabloc -->
    <remap from="~/input/yabloc/image" to="/localization/pose_estimator/yabloc/image_processing/undistorted/image_raw"/>
    <remap from="~/output/yabloc/image" to="/localization/pose_estimator/yabloc/image_processing/undistorted/image_raw/relay"/>
    <remap from="~/yabloc_suspend_srv" to="/yabloc_suspend_srv"/>
    <!-- eagleye -->
    <remap from="~/input/eagleye/pose_with_covariance" to="/localization/eagleye/pose_with_covariance/to_relay"/>
    <remap from="~/output/eagleye/pose_with_covariance" to="/localization/pose_estimator/pose_with_covariance"/>
    <!-- ndt -->
    <remap from="~/input/ndt/pointcloud" to="/sensing/lidar/top/outlier_filtered/pointcloud"/>
    <remap from="~/output/ndt/pointcloud" to="/sensing/lidar/top/outlier_filtered/pointcloud/relay"/>

    <!-- rule -->
    <remap from="~/input/pointcloud_map" to="/map/pointcloud_map"/>
    <remap from="~/input/vector_map" to="/map/vector_map"/>
    <remap from="~/input/pose_with_covariance" to="/localization/pose_with_covariance"/>
    <remap from="~/input/initialization_state" to="/localization/initialization_state"/>
  </node>
</launch>
